lib/https.js

```diff
@@ -180,6 +180,30 @@ Agent.prototype.getName = function getName(options) {
   if (options.secureProtocol)
     name += options.secureProtocol;
 
+  name += ':';
+  if (options.crl)
+    name += options.crl;
+
+  name += ':';
+  if (options.honorCipherOrder !== undefined)
+    name += options.honorCipherOrder;
+
+  name += ':';
+  if (options.ecdhCurve)
+    name += options.ecdhCurve;
+
+  name += ':';
+  if (options.dhparam)
+    name += options.dhparam;
+
+  name += ':';
+  if (options.secureOptions !== undefined)
+    name += options.secureOptions;
+
+  name += ':';
+  if (options.sessionIdContext)
+    name += options.sessionIdContext;
```

Comment:
Hm, this is kind of starting to look like working with a list of options might be an easier choice…

Reply:
Agreed. How about something like:

```js
const components = [
  options.ca,
  options.cert,
  options.ciphers,
  options.key,
  options.pfx,
  options.rejectUnauthorized,
  options.servername !== options.host ? options.servername : undefined,
  options.secureProtocol,
  options.crl,
  options.honorCipherOrder,
  options.ecdhCurve,
  options.dhparam,
  options.secureOptions,
  options.sessionIdContext
];

for (const component of components) {
  name += ':';
  if (component != null) {
    name += component;
  }
}

// or:
// name += components
//   .map(component => component != null ? `:${component}` : ':')
//   .join('');
```

There's a potential slight difference in serialization (`null`s wouldn't be added for `rejectUnauthorized`, `honorCipherOrder`, or `secureOptions` any more), but it seems like `null` would either have no impact or be invalid for all three so it feels alright to me.

Thoughts?

Reply:
> but it seems like `null` would either have no impact or be invalid for all three so it feels alright to me.

I agree.

Reply:
updated